Sources/NIO/Channel.swift

```diff
@@ -140,6 +140,30 @@ public protocol Channel: AnyObject, ChannelOutboundInvoker {
     ///
     /// - warning: Unsafe, this is for use in NIO's core only.
     var _channelCore: ChannelCore { get }
+
+    /// Returns a view of the `Channel` exposing synchronous versions of `setOption` and `getOption`.
+    /// The default implementation returns `nil`, and `Channel` implementations must opt in to
+    /// support this behavior.
+    var syncOptions: NIOSynchronousChannelOptions? { get }
+}
+
+extension Channel {
+    /// Default implementation: `NIOSynchronousChannelOptions` are not supported.
+    public var syncOptions: NIOSynchronousChannelOptions? {
```

Comment:
It's a shame we can't have opaque return types for this.

Reply:
Yup... life is much nicer in 5.2+

Reply:
@Lukasa / @glbrntt I mean we _could_ (not saying we should) make this a 5.2+ only feature and guard it...

Reply:
I suspect it isn't worth it, tbh. If we were really getting to the stage where we thought it was worth it I'd rather just be dropping support for 5.0 and 5.1.